multi-layer-app-config/README.md

```diff
@@ -40,6 +40,36 @@ We can extend App CR with on optional list of `ConfigMap`/`Secret` objects. Conf
 `userConfig:` properties (if given) is applied at the end of this list (to maintain backward compatibility).
 `app-operator` does merging of all layers on the list, from top to bottom, instead of just merging the two properties.
 
+This new list is called `extraConfigs`. Each entry in it has a field called the `priority`.
+It has a default value assumed that  makes them to be applied before `config` to keep the backward compatibility.
+On top of that `config` and `userConfig` gets a priority level - documented in App Platform - as well it becomes
+possible to apply some of the `extraConfigs` between the `config` and `userConfig` entries or even after `userConfig`.
+The bedrock is still considered to be what is in the catalog. It is not possible to apply `extraConfigs` before that.
+
+#### Merging algorithm
+
+Assuming the following priorities for the platform layers:
+
+- Catalog: A (e.g.: 0)
+- Cluster (`config`): B (e.g.: 50)
+- User (`userConfig`): (e.g.: 100)
+
+The distance (d) between each priority level should be the same.
+The `priority` field is validated on the CRD schema definition that it must be within range of: `[A, C + d]` and have
+the default value of: `A + d / 2` rounded up if necessary.
+
+The merging algorithm is as follows:
+
+1. Configuration from the catalog (A)
+2. All entries from `extraConfigs` with priority of P: A <= P <= B
+3. Configuration from `config` entry (B)
+4. All entries from `extraConfigs` with priority of P: B < P <= C
+5. Configuration from `userConfig` entry (C)
+6. All entries from `extraConfigs` with priority of P: C < P
+
+In case of multiple items in `extraConfigs` appear with the same priority then the item that appears first
```

Comment:
```suggestion
In case of multiple items in `extraConfigs` having the same priority, the order on the list is binding, with the item lower on the list being merged later (overriding those higher on the list)
```